include/moqt/client.h

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+
+#pragma once
+
+#include <moqt/config.h>
+#include <moqt/core/transport.h>
+
+namespace moq::transport {
+    using namespace qtransport;
+
+    /**
+     * @brief MoQT Client
+     *
+     * @details MoQT Client is the handler of the MoQT QUIC transport IP connection.
+     */
+    class Client : public Transport
+    {
+      public:
+        /**
+         * @brief MoQ Client Constructor to create the client mode instance
+         *
+         * @param cfg           MoQT Client Configuration
+         */
+        Client(const ClientConfig& cfg)
+          : Transport(cfg)
+        {
+        }
+
+        ~Client() = default;
+
+        /**
+         * @brief Starts a client connection via a transport thread (non-blocking)
+         *
+         * @details Makes a client connection session and runs in a newly created thread. All control and track
+         *   callbacks will be run based on events.
+         *
+         * @return Status indicating state or error. If successful, status will be
+         *    kClientConnecting.
+         */
+        Status Connect();
```

Comment:
Since connect's logic happens on a separate transport thread, is Status() is synchronous ? Also kClientConnecting doesn't mean successful though, isn't it ?